### PR TITLE
feat: add support for auth token file to be a json file

### PR DESCRIPTION
**Pull request by umglurf**

### CODE REVIEW

File: cmd/terraform-registry/main_test.go [hunk 1]
```diff
@@ -26,3 +26,21 @@ func TestParseAuthTokenFile(t *testing.T) {
 	is.Equal(tokens[1], "bar")
 	is.Equal(tokens[2], "baz")
 }
+
+func TestParseAuthTokenFileJson(t *testing.T) {
```
Comment (stigok) [inline cmd/terraform-registry/main_test.go:30]: Can you change the name of the other test to be suffixed with `Newlines` or similar? `TestParseAuthTokenFileNewlines`

Comment (stigok) [reply cmd/terraform-registry/main_test.go:30]: And please capitalise `JSON` in the test name

File: cmd/terraform-registry/main.go [hunk 2]
```diff
@@ -140,12 +141,22 @@ func parseAuthTokensFile(filepath string) ([]string, error) {
 	if err != nil {
 		return tokens, err
 	}
-
-	lines := strings.Split(string(b), "\n")
-	for _, token := range lines {
-		if token = strings.TrimSpace(token); token != "" {
+	if strings.HasSuffix(filepath, ".json") {
+		tokenmap := make(map[string]string)
+		err = json.Unmarshal([]byte(b), &tokenmap)
```
Comment (stigok) [inline cmd/terraform-registry/main.go:146]: `b` is already a byte slice

File: README.md [hunk 3]
```diff
@@ -32,11 +32,24 @@ All registry store types share some configuration in common.
...
 - `-auth-disabled`: Disable HTTP bearer token authentication (default: `false`)
-- `-auth-tokens-file`: File containing newline separated strings of valid tokens
+- `-auth-tokens-file`: File containing tokens. See below for details.
 - `-tls-enabled`: Whether to enable TLS termination (default: `false`)
 - `-tls-cert-file`: Path to TLS certificate file
 - `-tls-key-file`: Path to TLS certificate private key file
 
+### The tokens file
+
+The tokens file can be in one of two formats. It can be a json file, in which case the filename must end with `.json'.
```
Comment (stigok) [inline README.md:42]: _JSON_ in capital letters.

Comment (stigok) [reply README.md:42]: Missing backtick after `.json`

File: README.md [hunk 4]
```diff
@@ -32,11 +32,24 @@ All registry store types share some configuration in common.
...
+If it is not a json file, it is expected to be a file with newline separated strings of valid tokens.
+
+The json file must be a map of key names and token, for example
+```json
+{
+  "token for myapp": "some token",
```
Comment (stigok) [inline README.md:48]: Could we be more clear here for the example keys? We're not currently using them for anything, so maybe we could call the keys here for _description for my first token1_ or similiar?

**Review by stigok — CHANGES_REQUESTED** [review]
Sweet!

File: cmd/terraform-registry/main_test.go (hunk 1 above)
Comment (umglurf) [reply cmd/terraform-registry/main_test.go]: Updated

File: cmd/terraform-registry/main.go (hunk 2 above)
Comment (umglurf) [reply cmd/terraform-registry/main.go]: Updated

File: README.md (hunk 3 above)
Comment (umglurf) [reply README.md]: Updated

File: README.md (hunk 4 above)
Comment (umglurf) [reply README.md]: Updated

File: cmd/terraform-registry/main_test.go [hunk 5]
```diff
@@ -38,9 +39,10 @@ func TestParseAuthTokenFileJson(t *testing.T) {
 
 	tokens, err := parseAuthTokensFile(f.Name())
 	is.NoErr(err)
+	sort.Strings(tokens)
```
Comment (stigok) [inline cmd/terraform-registry/main_test.go:42]: Nice catch :)

File: README.md [hunk 6]
```diff
@@ -32,11 +32,24 @@ All registry store types share some configuration in common.
...
-- `-auth-tokens-file`: File containing newline separated strings of valid tokens
+- `-auth-tokens-file`: File containing tokens. See below for details.
 - `-tls-enabled`: Whether to enable TLS termination (default: `false`)
 - `-tls-cert-file`: Path to TLS certificate file
 - `-tls-key-file`: Path to TLS certificate private key file
 
+### The tokens file
+
+The tokens file can be in one of two formats. It can be a JSON file, in which case the filename must end with `.json`.
+If it is not a json file, it is expected to be a file with newline separated strings of valid tokens.
```
Comment (stigok) [inline README.md:43]: json -> JSON

**Review by stigok — CHANGES_REQUESTED** [review]

File: README.md (hunk 6 above)
Comment (umglurf) [reply README.md]: updated

**Review by stigok — APPROVED** [review]

**Pull request merged**